pkg/action/failback.go

```diff
@@ -0,0 +1,216 @@
+package action
+
+import (
+	"fmt"
+
+	storkv1 "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	"github.com/libopenstorage/stork/pkg/log"
+	"github.com/libopenstorage/stork/pkg/resourceutils"
+	"github.com/libopenstorage/stork/pkg/schedule"
+	"github.com/libopenstorage/stork/pkg/utils"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (ac *ActionController) verifyMigrationScheduleBeforeFailback(action *storkv1.Action) {
+
+	if action.Status.Status == storkv1.ActionStatusSuccessful {
+		action.Status.Stage = storkv1.ActionStageScaleDownDestination
+		action.Status.Status = storkv1.ActionStatusInitial
+		action.Status.FinishTimestamp = metav1.Now()
+		ac.updateAction(action)
```

Comment:
I see that updateAction method's call returns an error. Do we not need to handle that error??

Reply:
After updateAction , we anyway return in current cases.
Wherever the error needs to be handled, the respective code will get added at that time.

As of now, if it fails the stage remains same, so the stage will be rerun.